### ISSUE TITLE
Whether an array is required or not should be controlled by `ArraySchema.arraySchema.required` instead of `ArraySchema.schema.required`

### ISSUE THREAD

**Issue by garrychnca** [swagger-api/swagger-core#3438]
https://github.com/swagger-api/swagger-core/blob/6f6e420b50cedbfef192610bb469817d2a242103/modules/swagger-core/src/main/java/io/swagger/v3/core/jackson/SwaggerAnnotationIntrospector.java#L63-L65

According to the comments of `ArraySchema` annotation, `ArraySchema.schema` defines the model of the **items** within the array, while `ArraySchema.arraySchema` defines the model for the model of the array itself.

Hence the code should be changed to use:
```java
if (arraySchema.arraySchema().required()) {
    return arraySchema.arraySchema().required();
}
```

**Comment by frantuma** [comment]
can you provide an example of code and expected result?

**Comment by garrychnca** [comment]
For the below DTO class:

```java
public class Person {
    @ArraySchema(
        arraySchema = @Schema(required = true, description = "The person")
    )
    private List<String> addresses;
}
```

I expect the generated spec to have (but it is not):

```json
{
  "Person": {
    "required": [
      "addresses"
    ]
  }
}
```

I need to work around this by (note the addition `schema` attribute to the `@ArraySchema` annotation):

```java
public class Person {
    @ArraySchema(
        arraySchema = @Schema(description = "The person"),
        schema = @Schema(required = true)
    )
    private List<String> addresses;
}
```

**Comment by frantuma** [comment]
Right, thanks for reporting it! Fixed in #3478 (current behaviour left untouched for compatibility, possibly removed in further versions)
